crates/nu-cmd-base/src/arg_glob.rs

```diff
@@ -0,0 +1,115 @@
+// utilities for expanding globs in command arguments
+use nu_protocol::{ShellError, Spanned};
+use std::path::PathBuf;
+use wax::{Glob, LinkBehavior, WalkBehavior};
+
+// standard glob options to use for filesystem command arguments
+const FS_CMD_WALK_BEHAVIOR: WalkBehavior = WalkBehavior {
+    depth: usize::MAX,
+    link: LinkBehavior::ReadTarget,
+};
+
+// expand a glob into matching paths, using FS_CMD_WALK_BEHAVIOR
+//todo: see about returning WalkError or BuildError, or enabling miette and dealing with diagnostic
+pub fn arg_glob(
+    pattern: &Spanned<String>, // the glob
+    include_dirs: bool,        // include dirs in results.  Default (f) only include files
+    cwd: &PathBuf,             // current working directory
+) -> Result<Vec<PathBuf>, ShellError> {
+    // eventually, windows only, and with better performance
+    // transform something like: C:\Users\me\.../{abc,def}*.txt to C\:/Users/me/.../{abc,def}*.txt
+    // such a string has good chance of working as a rooted glob pattern on Windows
+    // User will have to quote metachars with [c] rather than \c.
+    let processed_pattern = pattern
+        .item
+        .replace('\\', "/")
+        .replace(r#":/"#, r#"\:/"#) // must come after above
+        .replace('(', r#"\("#) // when path includes Program Files (x86)
+        .replace(')', r#"\)"#);
+
+    let (prefix, glob) = match Glob::new(&processed_pattern) {
+        Ok(p) => p.partition(),
+        Err(e) => {
+            return Err(ShellError::InvalidGlobPattern(e.to_string(), pattern.span));
+        }
+    };
+
+    let path = match nu_path::canonicalize_with(prefix, cwd) {
+        Ok(path) => path,
+        Err(e) if e.to_string().contains("os error 2") =>
+        // path we're trying to glob doesn't exist,
+        {
+            PathBuf::new() // ensure glob doesn't match anything
+        }
+        Err(e) => return Err(ShellError::ErrorExpandingGlob(format!("{e}"), pattern.span)),
+    };
+
+    let mut rv: Vec<PathBuf> = Vec::new();
+    for w in glob
+        .walk_with_behavior(path, FS_CMD_WALK_BEHAVIOR)
```

Comment:
Since depth is usize::MAX, is there exposure here that we could have performance problems because it's searching for deeply nested matches?

Reply:
Yes.  But it's the same depth currently used by `glob` command.  Does linux `cp` have any limit?

Reply:
I have no idea what the limitations of `cp` may be.

Another thing I was thinking was, since the depth is set to max, is there the possibility of false positive completions here? like you do something with the {} syntax, intending to change files in the current folder, but some 10 levels deeper there are things that match.

I'm not suggesting that you change the depth, I'm just trying to think if there are ways it can fail. I think I originally wrote glob and made it take the default max depth to begin with.

Reply:
Seems the various OS filesystems have depth limits much less than glob,
https://stackoverflow.com/questions/55292054/what-is-the-cause-of-the-hard-limit-on-the-directory-nesting-depth-returned-by-g
https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2003/cc781134(v=ws.10)?redirectedfrom=MSDN#maximum-sizes-on-an-ntfs-volume

User will hit these limits before any default we put into glob.  

I'm not worried about false positives _per se_.  Any unexpected results would be files that turned out to match the glob the user specified, even if that's not what the user thought the glob would do.